### PR TITLE
refactor($compile): remove unnecessary and invalid $destroyed listener

### PR DESCRIPTION
**Pull request by jbedard**
I noticed when bindToController is used the scope `$destroyed` event is listened for, but it should be `$destroy`.  But looking further into it I don't think we need to listen for any destroy event in this case so this PR just deletes it.

### CODE REVIEW

File: src/ng/compile.js [hunk 1]
```diff
@@ -2644,16 +2639,12 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
...
       });
-      var destroyBindings = onNewScopeDestroyed ? function destroyBindings() {
+
+      return onNewScopeDestroyed && function destroyBindings() {
         for (var i = 0, ii = onNewScopeDestroyed.length; i < ii; ++i) {
           onNewScopeDestroyed[i]();
         }
-      } : noop;
-      if (newScope && destroyBindings !== noop) {
-        newScope.$on('$destroy', destroyBindings);
```
Comment (jbedard) [inline src/ng/compile.js]: I moved this out to the 1 of 3 `initializeDirectiveBindings` invocations that actually need it. I figured now that `initializeDirectiveBindings` binds to any `destination` and not specifically a scope this logic no longer belonged here...

**Comment by lgalfaso** [comment]
@petebacondarwin WDYT?

File: src/ng/compile.js [hunk 2]
```diff
@@ -1985,24 +1978,28 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
           compile.$$addScopeClass($element, true);
           isolateScope.$$isolateBindings =
               newIsolateScopeDirective.$$isolateBindings;
-          initializeDirectiveBindings(scope, attrs, isolateScope,
-                                      isolateScope.$$isolateBindings,
-                                      newIsolateScopeDirective, isolateScope);
+          var parentWatchDestoryer = initializeDirectiveBindings(scope, attrs, isolateScope,
```
Comment (Narretz) [inline src/ng/compile.js]: -> *Destroyer* (also on line 1984 and 1985)

**Comment by Narretz** [comment]
I think the commit message should be more explicit: why exactly do we not need to listen to $destroy in this case?

**Comment by jbedard** [comment]
Fixed the var name typo and rebased. I'll have to look into it again to remember why the listener isn't actually needed...

**Comment by petebacondarwin** [comment]
I tweaked, embellished and merged. Thanks @jbedard - do you just spend your evenings scouring the compiler??

**Pull request closed without merging**